apps/bundle-size/src/utils/buildFixture.js

```diff
@@ -0,0 +1,148 @@
+
+const { hrToSeconds } = require('./helpers');
+
+/**
+ * @param {string} fixturePath
+ * @param {string} outputPath
+ *
+ * @return {import("webpack").Configuration}
+ */
+function createWebpackConfig(fixturePath, outputPath) {
+  return {
+    name: 'client',
+    target: 'web',
+    mode: 'production',
+
+    cache: {
+      type: 'memory',
+    },
+    externals: {
+      react: 'React',
+      'react-dom': 'ReactDOM',
+    },
+
+    entry: fixturePath,
+    output: {
+      filename: path.basename(outputPath),
+      path: path.dirname(outputPath),
+
+      pathinfo: true,
+    },
+    performance: {
+      hints: false,
+    },
+    optimization: {
+      minimize: false,
+    },
+    stats: {
+      optimizationBailout: true,
+    },
+  };
+}
+
+/**
+ * @param {import("webpack").Configuration} webpackConfig
+ * @return {Promise<null>}
+ */
+function webpackAsync(webpackConfig) {
+  return new Promise((resolve, reject) => {
+    const compiler = webpack(webpackConfig);
+
+    compiler.run(err => {
+      if (err) {
+        reject(err);
+      }
+
+      resolve(null);
+    });
+  });
+}
+
+// ---
+
+/** @typedef {{ name: string, path: string, minifiedSize: number, gzippedSize: number }} BuildResult */
+
+/**
+ * Builds a fixture with Webpack and then minifies it with Terser. Produces two files as artifacts:
+ * - partially minified file (.output.js) for debugging
+ * - fully minified file (.min.js)
+ *
+ * @param {import('./prepareFixture').PreparedFixture} preparedFixture
+ * @param {boolean} verbose
+ *
+ * @return {Promise<BuildResult>}
+ */
+module.exports = async function buildFixture(preparedFixture, verbose) {
+  const webpackStartTime = process.hrtime();
+
+  const webpackOutputPath = preparedFixture.absolutePath.replace(/.fixture.js$/, '.output.js');
+  const config = createWebpackConfig(preparedFixture.absolutePath, webpackOutputPath);
+
+  await webpackAsync(config);
+
+  if (verbose) {
+    console.log(
+      [
+        chalk.blue('[i]'),
+        `"${path.basename(preparedFixture.relativePath)}": Webpack in ${hrToSeconds(process.hrtime(webpackStartTime))}`,
+      ].join(' '),
+    );
+  }
+
+  // ---
+
+  const terserStartTime = process.hrtime();
+  const terserOutputPath = preparedFixture.absolutePath.replace(/.fixture.js$/, '.min.js');
+
+  const webpackOutput = (await fs.promises.readFile(webpackOutputPath)).toString();
+
+  const [terserOutput, terserOutputMinified] = await Promise.all([
+    // Performs only dead-code elimination
+    /* eslint-disable @typescript-eslint/naming-convention */
+    minify(webpackOutput, {
+      mangle: false,
+      output: {
+        beautify: true,
+        comments: true,
+        preserve_annotations: true,
+      },
+    }),
+    minify(webpackOutput, {
+      output: {
+        comments: false,
+      },
+    }),
+    /* eslint-enable @typescript-eslint/naming-convention */
+  ]);
+
+  await fs.promises.writeFile(webpackOutputPath, terserOutput.code);
```

Comment:
note: promises api is supported natively since node 10, any reason why not use it ?

Reply:
Do you mean this 👇?

```diff
-const fs = require('fs')
+const fs = require('fs/promises')
```

```diff
-await fs.promises.writeFile(webpackOutputPath, terserOutput.code);
+await fs.writeFile(webpackOutputPath, terserOutput.code);
```

Reply:
I removed all usages of `fs-extra` in 43e09ba.